scheduler/workflow.go

```diff
@@ -335,26 +336,38 @@ func (s *schedulerWorkflow) StateString() string {
 }
 
 func (s *schedulerWorkflow) workJobs(prs []*processNode, pus []*publishNode, t *task, pj job) {
+	// create a waitgroup track pending goroutines submitting jobs
+	wg := &sync.WaitGroup{}
 	for _, pr := range prs {
-		j := newProcessJob(pj, pr.Name(), pr.Version(), pr.InboundContentType, pr.config.Table(), t.metricsManager, t.id)
-		errors := t.manager.Work(j).Promise().Await()
-		if len(errors) != 0 {
-			t.failedRuns++
-			t.lastFailureTime = t.lastFireTime
-			t.lastFailureMessage = errors[len(errors)-1].Error()
-			return
-		}
+		go func() {
+			wg.Add(1)
+			defer wg.Done()
+			j := newProcessJob(pj, pr.Name(), pr.Version(), pr.InboundContentType, pr.config.Table(), t.metricsManager, t.id)
+			errors := t.manager.Work(j).Promise().Await()
```

Comment:
See usage of `AndThen` here: https://github.com/intelsdi-x/snap/blob/master/pkg/promise/promise_test.go#L56